Master fails to compile

Describe your bug report or feature request here
Compile fails, I have piggy backed off the AUR of libresprite and have done the following:

  mkdir -p build && cd build
  git submodule update --init --recursive
  cmake -DUSE_SHARED_PIXMAN=ON \
    -DWITH_WEBP_SUPPORT=ON \
    -DUSE_SHARED_LIBWEBP=ON \
    -DUSE_SHARED_CURL=ON \
    -DUSE_SHARED_GIFLIB=ON \
    -DUSE_SHARED_JPEGLIB=ON \
    -DUSE_SHARED_ZLIB=ON \
    -DUSE_SHARED_LIBPNG=ON \
    -DUSE_SHARED_LIBLOADPNG=OFF \
    -DUSE_SHARED_TINYXML=ON \
    -DENABLE_UPDATER=OFF \
    -DUSE_SHARED_FREETYPE=ON \
    -DFREETYPE_INCLUDE_DIR=/usr/include/freetype2 \
    -DCMAKE_INSTALL_PREFIX:STRING=/usr ..
make libresprite

### LibreSprite and System version

* LibreSprite (version number or commit hash): 0a01bb9 
* System (Win/OSX/Linux, version, distribution): Linux, Arch

[ 34%] Building CXX object src/she/CMakeFiles/she.dir/alleg4/alleg_display.cpp.o
In file included from /usr/include/freetype2/freetype/freetype.h:33,
                 from /usr/include/freetype2/freetype/ftglyph.h:37,
                 from /home/godofgrunts/temp/LibreSprite/src/./ft/freetype_headers.h:12,
                 from /home/godofgrunts/temp/LibreSprite/src/./ft/face.h:13,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/common/freetype_font.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/common/generic_surface.h:12,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/alleg4/alleg_surface.h:14,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:18:
/usr/include/freetype2/freetype/config/ftconfig.h:63: warning: "HAVE_STDINT_H" redefined
 #define HAVE_STDINT_H 1
 
In file included from /home/godofgrunts/temp/LibreSprite/src/./base/base.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/./config.h:40,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:8:
/home/godofgrunts/temp/LibreSprite/build/src/base/base/config.h:11: note: this is the location of the previous definition
 #define HAVE_STDINT_H
 
In file included from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/base.h:41,
                 from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro.h:25,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/alleg4/alleg_surface.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:18:
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/alcompat.h:44:22: error: conflicting declaration of C function ‘fixed fadd(fixed, fixed)’
       AL_ALIAS(fixed fadd(fixed x, fixed y), fixadd(x, y))
                      ^~~~
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/internal/alconfig.h:164:49: note: in definition of macro ‘AL_ALIAS’
       static __attribute__((unused)) __inline__ DECL    \
                                                 ^~~~
In file included from /usr/include/c++/8.2.0/cmath:45,
                 from /usr/include/c++/8.2.0/math.h:36,
                 from /home/godofgrunts/temp/LibreSprite/src/./base/base.h:13,
                 from /home/godofgrunts/temp/LibreSprite/src/./config.h:40,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:8:
/usr/include/bits/mathcalls-narrow.h:24:20: note: previous declaration ‘float fadd(double, double)’
 __MATHCALL_NARROW (__MATHCALL_NAME (add), __MATHCALL_REDIR_NAME (add), 2);
                    ^~~~~~~~~~~~~~~
In file included from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/base.h:41,
                 from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro.h:25,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/alleg4/alleg_surface.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:18:
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/alcompat.h:45:22: error: conflicting declaration of C function ‘fixed fsub(fixed, fixed)’
       AL_ALIAS(fixed fsub(fixed x, fixed y), fixsub(x, y))
                      ^~~~
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/internal/alconfig.h:164:49: note: in definition of macro ‘AL_ALIAS’
       static __attribute__((unused)) __inline__ DECL    \
                                                 ^~~~
In file included from /usr/include/c++/8.2.0/cmath:45,
                 from /usr/include/c++/8.2.0/math.h:36,
                 from /home/godofgrunts/temp/LibreSprite/src/./base/base.h:13,
                 from /home/godofgrunts/temp/LibreSprite/src/./config.h:40,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:8:
/usr/include/bits/mathcalls-narrow.h:33:20: note: previous declaration ‘float fsub(double, double)’
 __MATHCALL_NARROW (__MATHCALL_NAME (sub), __MATHCALL_REDIR_NAME (sub), 2);
                    ^~~~~~~~~~~~~~~
In file included from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/base.h:41,
                 from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro.h:25,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/alleg4/alleg_surface.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:18:
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/alcompat.h:46:22: error: conflicting declaration of C function ‘fixed fmul(fixed, fixed)’
       AL_ALIAS(fixed fmul(fixed x, fixed y), fixmul(x, y))
                      ^~~~
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/internal/alconfig.h:164:49: note: in definition of macro ‘AL_ALIAS’
       static __attribute__((unused)) __inline__ DECL    \
                                                 ^~~~
In file included from /usr/include/c++/8.2.0/cmath:45,
                 from /usr/include/c++/8.2.0/math.h:36,
                 from /home/godofgrunts/temp/LibreSprite/src/./base/base.h:13,
                 from /home/godofgrunts/temp/LibreSprite/src/./config.h:40,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:8:
/usr/include/bits/mathcalls-narrow.h:30:20: note: previous declaration ‘float fmul(double, double)’
 __MATHCALL_NARROW (__MATHCALL_NAME (mul), __MATHCALL_REDIR_NAME (mul), 2);
                    ^~~~~~~~~~~~~~~
In file included from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/base.h:41,
                 from /home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro.h:25,
                 from /home/godofgrunts/temp/LibreSprite/src/./she/alleg4/alleg_surface.h:11,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:18:
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/alcompat.h:47:22: error: conflicting declaration of C function ‘fixed fdiv(fixed, fixed)’
       AL_ALIAS(fixed fdiv(fixed x, fixed y), fixdiv(x, y))
                      ^~~~
/home/godofgrunts/temp/LibreSprite/src/allegro/include/allegro/internal/alconfig.h:164:49: note: in definition of macro ‘AL_ALIAS’
       static __attribute__((unused)) __inline__ DECL    \
                                                 ^~~~
In file included from /usr/include/c++/8.2.0/cmath:45,
                 from /usr/include/c++/8.2.0/math.h:36,
                 from /home/godofgrunts/temp/LibreSprite/src/./base/base.h:13,
                 from /home/godofgrunts/temp/LibreSprite/src/./config.h:40,
                 from /home/godofgrunts/temp/LibreSprite/src/she/alleg4/alleg_display.cpp:8:
/usr/include/bits/mathcalls-narrow.h:27:20: note: previous declaration ‘float fdiv(double, double)’
 __MATHCALL_NARROW (__MATHCALL_NAME (div), __MATHCALL_REDIR_NAME (div), 2);
                    ^~~~~~~~~~~~~~~
make[3]: *** [src/she/CMakeFiles/she.dir/build.make:76: src/she/CMakeFiles/she.dir/alleg4/alleg_display.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:1195: src/she/CMakeFiles/she.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:176: src/CMakeFiles/libresprite.dir/rule] Error 2
make: *** [Makefile:175: libresprite] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,6 +300,7 @@ set(PLATFORM_LIBS)
 # Allegro 4 backend
 if(USE_ALLEG4_BACKEND)
   add_definitions(-DUSE_ALLEG4_BACKEND)
+  add_definitions(-DALLEGRO_NO_FIX_ALIASES)
 
   if(USE_SHARED_ALLEGRO4)
     # Find the shared Allegro 4 library

